animated-transformer/src/lib/gtensor/gtensor.ts

```diff
@@ -799,39 +807,41 @@ export class GTensor<G extends DName> {
     return new GTensor<G>(this.tensor.equal(g2.tensor), this.dimNames);
   }
 
-  // TODO: support batched gather.
-  public gather<D extends G, G2 extends DName>(
+  public gather<D extends G, B extends G & G2, G2 extends DName>(
```

Comment:
Add a comment with a TODO and create a bug on tfjs to fix grad for a number of batch dimensions higher than one